addon/components/horizontal-bar-chart.js

```diff
@@ -451,12 +455,35 @@ const HorizontalBarChartComponent = ChartComponent.extend(FloatingTooltipMixin,
         return this._getElementForValue(valueLabelElements, val);
       });
 
-      const leftLabels = negativeValueLabels.concat(positiveGroupingLabels);
-      const rightLabels = positiveValueLabels.concat(negativeGroupingLabels);
+      const maxNegativeValueLabelWidth = this._maxWidthOfElements(negativeValueLabels);
+      const maxPositiveGroupingLabelWidth = this._maxWidthOfElements(positiveGroupingLabels);
+
+      const maxPositiveValueLabelWidth = this._maxWidthOfElements(positiveValueLabels);
+      const maxNegativeGroupingLabelWidth = this._maxWidthOfElements(negativeGroupingLabels);
+
+      const padding = this.get('labelPadding');
+      const minValue = this.get('minValue');
+      const maxValue = this.get('maxValue');
+      const width = this.get('outerWidth') - maxNegativeValueLabelWidth - maxPositiveValueLabelWidth - (padding * 2);
+      const xScale = this.xScaleForWidth(width);
+
+      const maxNegativeBarWidth = Math.abs(xScale(minValue) - xScale(0));
+      const maxPositiveBarWidth = Math.abs(xScale(maxValue) - xScale(0));
+
+      var leftWidth;
+      if (maxNegativeBarWidth + maxNegativeValueLabelWidth > maxPositiveGroupingLabelWidth) {
+        leftWidth = maxNegativeValueLabelWidth;
+      } else {
+        leftWidth = maxPositiveGroupingLabelWidth - maxNegativeBarWidth;
+      }
+
+      var rightWidth;
+      if (maxPositiveBarWidth + maxPositiveValueLabelWidth > maxNegativeGroupingLabelWidth) {
+        rightWidth = maxPositiveValueLabelWidth;
+      } else {
+        rightWidth = maxNegativeGroupingLabelWidth - maxPositiveBarWidth;
```

Comment:
@philpee2 I tried this logic to begin with too, but this doesn't work unfortunately. The bar widths will change when you change the margin and hence will shift the axis.